### PR TITLE
remove logging based step events

### PR DESCRIPTION
**Pull request by alangenfeld**
consolidate down to just one way of communicating step execution events, removing the logging variant and add `success` and `start` to the yield based approach

### CODE REVIEW

**Comment by alangenfeld** [comment]
not quite ready for review, but up for preview

**Comment by schrockn** [comment]
I think we should discuss this a bit as this will break dagit as currently implemented (i believe) since dagit relies on these events coming through the event_callback, which comes from the logging channel.

File: python_modules/dagster/dagster/core/execution_plan/simple_engine.py [hunk 1]
```diff
@@ -219,20 +216,11 @@ def _execute_steps_core_loop(step_context, inputs, intermediates_manager):
 
             yield _create_step_event(step_context, step_output_value, intermediates_manager)
 
-        step_context.events.execution_plan_step_success(step.key, timer_result.millis)
+        yield ExecutionStepEvent.step_success_event(
+            step_context, StepSuccessData(time_ms=timer_result.millis)
+        )
 
     except DagsterError as dagster_error:
```
Comment (schrockn) [inline python_modules/dagster/dagster/core/execution_plan/simple_engine.py]: kill this totally

File: python_modules/dagster/dagster_tests/core_tests/test_event_logging.py [hunk 2]
```diff
@@ -76,24 +86,24 @@ def _event_callback(record):
     assert result.success
     assert events
 
-    start_event = single_event(events, EventType.EXECUTION_PLAN_STEP_START)
+    start_event = single_dagster_event(events, ExecutionStepEventType.STEP_START)
     assert start_event.pipeline_name == 'single_solid_pipeline'
-    assert start_event.solid_name == 'solid_one'
+    assert start_event.dagster_event.solid_name == 'solid_one'
```
Comment (schrockn) [inline python_modules/dagster/dagster_tests/core_tests/test_event_logging.py]: make pipeline_name available at the other level too

File: python_modules/dagster/dagster_tests/core_tests/test_event_logging.py [hunk 3]
```diff
@@ -76,24 +86,24 @@ def _event_callback(record):
...
+    assert start_event.dagster_event.solid_definition_name == 'solid_one'
```
Comment (schrockn) [inline python_modules/dagster/dagster_tests/core_tests/test_event_logging.py]: Soonish in a follow on diff I think we should have a top level event that is either typed LogMessage or DagsterEvent with a small shared interface. e.g. (LogMessage, DagsterEvent)

**Comment by alangenfeld** [comment]
archive

**Pull request closed without merging**